refactor(compiler): Add a new helper method `getOwningNgModule`.

dylhunn opened the pull request:
This helper accepts a class for an Angular trait, and returns the NgModule which owns that trait. This will be useful for the language service import project, which needs to edit import arrays on the module.

alxhub reviewed (approved):
As with the others, this shouldn't be a `feat` as it's not a public-facing feature of Angular.

dylhunn commented on packages/compiler-cli/test/ngtsc/env.ts:
```diff
@@ -271,6 +272,17 @@ export class NgtscTestEnvironment {
     return defaultGatherDiagnostics(program as api.Program) as ts.Diagnostic[];
   }
 
+  driveTemplateTypeChecker(): {program: ts.Program, checker: TemplateTypeChecker} {
```
Ack, I'll do this next time.

dylhunn committed:
refactor(compiler): Add a new helper method `getOwningNgModule`.

This helper accepts a class for an Angular trait, and returns the NgModule which owns that trait. This will be useful for the language service import project, which needs to edit import arrays on the module.

dylhunn commented:
All comments resolved. Since this is going to need a rebase, I will wait for [the other PR](https://github.com/angular/angular/pull/47180) to merge first before adding `action: merge`.

dylhunn commented:
merge-assistance: merge #47180 first; this PR is rebased on that PR

alxhub commented:
This PR was merged into the repository by commit edf7f670f06da8dc7c6a7de8e6b9366e794bf0ea.
